File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Budgets/Commands/EditSubmitBudgetCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.EarlierStage.Budgets;
using SIE.Web.Command;
using System;

namespace SIE.Web.EMS.EarlierStage.Budgets.Commands
{
    /// <summary>
    /// 编辑界面-提交预算
    /// </summary>
    [JsCommand("SIE.Web.EMS.EarlierStage.Budgets.Commands.EditSubmitBudgetCommand")]
    public class EditSubmitBudgetCommand : FormSaveCommand
    {
        /// <summary>
        /// 编辑界面-提交预算
        /// </summary>
        /// <param name="entity"></param>
        protected override void DoSave(Entity entity)
        {
            var budget = entity as Budget;
            if (budget == null)
            {
                throw new ValidationException("提交预算执行失败，数据错误".L10N());
            }

            RT.Service.Resolve<BudgetController>().EditSubmitBudget(budget);
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Budgets/Commands/EditSubmitBudgetCommand.js
SIE.defineCommand('SIE.Web.EMS.EarlierStage.Budgets.Commands.EditSubmitBudgetCommand', {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "保存并提交", group: "edit", iconCls: "icon-Refuse icon-blue" },
    canExecute: function (view) {
        var current = view.getCurrent();

        if (!current) return false;

        //审核状态：10 待提交 50 驳回
        if (current.data.ApprovalStatus !== 10 && current.data.ApprovalStatus !== 50) {
            return false;
        }

        return current.isDirty();
    },
    /**
 *  保存后事件
 * @param {} view 
 * @returns {} 
 */
    onSaved: function (view, res) {
        var current = view.getCurrent();
        current.markSaved();
        SIE.Msg.showInstantMessage('保存并提交成功！'.t());
        window.setTimeout(function () {
            CRT.Event.fire(view.model + "_refresh", current.data.Id);
            CRT.Workbench.closeCurrentTab();
        }, 3000);
    }
});